### PR TITLE
Notebook Lifecycle Config create, describe and delete

### PR DESCRIPTION
**Pull request by jweite**
Re Issue #3416.

### CODE REVIEW

File: moto/sagemaker/responses.py [hunk 1]
```diff
@@ -239,3 +239,38 @@ def delete_training_job(self):
...
+                ),
+                on_create=self._get_param("OnCreate"),
+                on_start=self._get_param("OnStart"),
+            )
+            response = {
+                "NotebookInstanceLifecycleConfigArn": lifecycle_configuration.notebook_instance_lifecycle_config_arn,
+            }
+            return 200, {}, json.dumps(response)
+        except AWSError as err:
+            return err.response()
```
Comment (bblommers) [inline moto/sagemaker/responses.py:258]: Is the try-except necessary? I don't see any errors being raised in models.py

File: moto/sagemaker/models.py [hunk 2]
```diff
@@ -677,6 +718,58 @@ def get_notebook_instance_tags(self, arn):
...
+            del self.notebook_instance_lifecycle_configurations[
+                notebook_instance_lifecycle_config_name
+            ]
+        except KeyError:
+            message = "Unable to delete Notebook Instance Lifecycle Config '{}'. (Details: Notebook Instance Lifecycle Config does not exist.".format(
+                FakeSageMakerNotebookInstanceLifecycleConfig.arn_formatter(
+                    notebook_instance_lifecycle_config_name, self.region_name
+                )
+            )
+            raise RESTError(
```
Comment (bblommers) [inline moto/sagemaker/models.py:767]: Let's extract both RESTerror into a ValidationError in exceptions.py, and

File: moto/sagemaker/models.py [hunk 3]
```diff
@@ -677,6 +718,58 @@ def get_notebook_instance_tags(self, arn):
...
+
+    def delete_notebook_instance_lifecycle_config(
+        self, notebook_instance_lifecycle_config_name
+    ):
+        try:
+            del self.notebook_instance_lifecycle_configurations[
+                notebook_instance_lifecycle_config_name
+            ]
+        except KeyError:
+            message = "Unable to delete Notebook Instance Lifecycle Config '{}'. (Details: Notebook Instance Lifecycle Config does not exist.".format(
```
Comment (bblommers) [inline moto/sagemaker/models.py:762]: Can we get a unit test for this scenario?

**Review by bblommers — CHANGES_REQUESTED** [review]

File: moto/sagemaker/responses.py (hunk 1 above)
Comment (jweite) [reply moto/sagemaker/responses.py:258]: Ah, copy-paste from another operation.  But on further thought, there is an important exception I need to be throwing on create: already exists.  So I'm leaving the try/except in place here and adding code to detect duplicate name and raise in the backend method.

File: moto/sagemaker/models.py (hunk 2 above)
Comment (jweite) [reply moto/sagemaker/models.py:767]: Done, and changed all existing raises of RESTError in this module to use it.

File: moto/sagemaker/models.py (hunk 3 above)
Comment (jweite) [reply moto/sagemaker/models.py:762]: Done.

**Review by bblommers — APPROVED** [review]
LGTM. Thanks @jweite!

**Pull request merged**